uaclient/tests/test_gpg.py

```diff
@@ -51,10 +52,11 @@ def test_export_single_key_from_keyring_dir(self, home_dir, tmpdir):
         # Create keyring with both ESM and CC-EAL2 keys
         source_key1.write(data.GPG_KEY1, "wb")
         source_key2.write(data.GPG_KEY2, "wb")
-        gpg.export_gpg_key(
-            source_keyfile=source_key1.strpath,
-            destination_keyfile=destination_keyfile,
-        )
+        with mock.patch("uaclient.util._subp", side_effect=_subp):
+            gpg.export_gpg_key(
```

Comment:
Sorry, just to double check if I understand the logic here. The fixture `_ subp` is yielding the original `_subp` function, therefore if we have a test that does need subp, it can use. Since we are adding the return of `_subp` here as a `side_effect` of the mock, can I assume that all those test that have that mechanism require the real `_subp` call ?

Reply:
@lucasmoura you are exactly right!

Arguably, we should change the code such that these unittests don't require subp, but that would require more significant changes than I thought made sense in this commit (e.g. re-writing this gpg test, re-writing the version_cmp function)